core/common/src/main/java/alluxio/util/proto/ProtoUtils.java

```diff
@@ -86,6 +87,39 @@ public static boolean isTruncatedMessageException(IOException e) {
     return e.getMessage().equals(truncatedMessage);
   }
 
+  /**
+   * @param acl {@link AccessControlList} to convert to protobuf
+   * @return new protobuf representation of the ACL
+   */
+  public static Acl.NewAccessControlList toProtoNew(AccessControlList acl) {
+    Acl.NewAccessControlList.Builder builder = Acl.NewAccessControlList.newBuilder();
+    builder.setOwningGroup(acl.getOwningGroup());
+    builder.setOwningUser(acl.getOwningUser());
+    builder.setMode(acl.getMode());
+
+    ExtendedACLEntries extended = acl.getExtendedEntries();
+    if (extended != null) {
+      builder.setMaskActions(extended.getMask().toModeBits().toProto());
+      builder.addAllNamedActions(() -> extended.getNamedUserActions().entrySet().stream().map(
+          entry -> Acl.ExtendedAclAction.newBuilder().setName(entry.getKey())
+              .setModeBits(entry.getValue().toModeBits().toProto()).build()).iterator());
+      builder.addAllGroupedActions(() -> extended.getNamedGroupActions().entrySet().stream().map(
+          entry -> Acl.ExtendedAclAction.newBuilder().setName(entry.getKey())
+              .setModeBits(entry.getValue().toModeBits().toProto()).build()).iterator());
+    }
+
+    if (acl instanceof DefaultAccessControlList) {
+      DefaultAccessControlList defaultAcl = (DefaultAccessControlList) acl;
+      builder.setIsDefault(true);
+      builder.setIsEmpty(defaultAcl.isEmpty());
+    }  else {
+      builder.setIsDefault(false);
+      // non default acl is always not empty
+      builder.setIsEmpty(false);
+    }
```

Comment:
This is duplicate from lines 149-157. Refactor into a shared function.

Reply:
i have a different opinion here. if we are going to deprecate the old one, it's okay to have some duplication between the old and new, to prevent the refactoring creating new bugs.